### ISSUE TITLE
 Possible misuse of aria-label - Invalid warning

### ISSUE THREAD

**Issue by Stefany93** [validator/validator#904]
`aria-label` is an optimal technique used to add more information about a list to screen readers. 
Shouldn't be flagged.

Reference - https://fae.disability.illinois.edu/rulesets/LIST_2/ 

![image](https://user-images.githubusercontent.com/2403939/73314304-24e3f900-41f3-11ea-986b-4afb0e43248c.png)

**Comment by sideshowbarker** [comment]
Thanks for taking time to report this.

The `aria-label` warning the checker is reporting  is based on that guidance in the following sources:

* https://www.w3.org/TR/using-aria/#practical-support-aria-label-aria-labelledby-and-aria-describedby
* https://developer.paciellogroup.com/blog/2017/07/short-note-on-aria-label-aria-labelledby-and-aria-describedby/

Neither of those sources recommends using `aria-label` being used with `ul` — which is why the HTML checker is reporting a warning fo it.

If you believe that `aria-label` should be allowed for `ul`, please post a comment at https://github.com/w3c/html-aam/issues/160.

It’s possible the *“Use the aria-label attribute to add a label to a list container element to provide an explicit text description of its contents”* guidance at https://fae.disability.illinois.edu/rulesets/LIST_2/ may need to be updated to align with the guidance in https://www.w3.org/TR/using-aria/#practical-support-aria-label-aria-labelledby-and-aria-describedby.

**Comment by Stefany93** [comment]
> Don't use aria-label or aria-labelledby on any other non-interactive content such as p, legend, li, or ul, because it is ignored.
https://www.w3.org/TR/using-aria/#practical-support-aria-label-aria-labelledby-and-aria-describedby

@sideshowbarker  I feel silly for reporting, I am sorry. Thank you for taking the time to correct me.

**Comment by sideshowbarker** [comment]
Don’t feel silly. It’s complicated. That’s why we added the warning to the checker — to help get feedback about how web author-developers are actually using `aria-label` in practice, and maybe to eventually get some additional requirements added to the ARIA spec so that’ll help author-developers get it all right. So it’s good that you took time to raise an issue here. It all helps.

**Comment by sandrina-p** [comment]
I crossed with the same warning. So, if I want to describe the content of a list, what's the correct usage then?

**Comment by sideshowbarker** [comment]
@sandrina-p Glad you got a solution. I’m not an accessibility domain expert so I can’t really speak with knowledge about what ARIA markup should be used for list containers — but from what I’ve gleaned by reading the discussions among the people involved in work on the ARIA spec who are knowledgeable about this, I think the main concern is around how AT/screen-readers handle the ARIA markup in practice. And I think it might be that some AT/SRs don’t actually voice ARIA markup for list containers in a way that, from a UX/usability point of view, is actually good for AT users.

Or I could be wrong and I’m misunderstanding the discussions.

But anyway I can say that I know the general recommendation is, no matter what ARIA markup you choose, you should actually test it in AT/SRs to make sure it’s having the effect in practice that you intend for it to — and especially not some other effect that inadvertently degrades UX for AT users

**Comment by sandrina-p** [comment]
@sideshowbarker yes, that's what I usually alwayd so: Test with a SR (Voice Over) and verify Accessibility tree on chrome. And using `aria-labelledby`, it works fine :)

![image](https://user-images.githubusercontent.com/14869087/74081765-83f30a80-4a4a-11ea-9304-4705eade71e3.png)

**Comment by sideshowbarker** [comment]
> @sideshowbarker yes, that's what I usually alwayd so: Test with a SR (Voice Over) and verify Accessibility tree on chrome. And using `aria-labelledby`, it works fine :)

Cool :)
